src/borg/remote.py

```diff
@@ -137,18 +210,29 @@ def serve(self):
                             logging.error(msg)
                             logging.log(tb_log_level, tb)
                         exc = "Remote Exception (see remote log for the traceback)"
-                        os.write(stdout_fd, msgpack.packb((1, msgid, e.__class__.__name__, exc)))
+                        if dictFormat:
+                            os.write(stdout_fd, msgpack.packb({b'i': msgid, b'exception_class': e.__class__.__name__, b'exception_args': e.args}))
+                        else:
+                            os.write(stdout_fd, msgpack.packb((1, msgid, e.__class__.__name__, exc)))
                     else:
-                        os.write(stdout_fd, msgpack.packb((1, msgid, None, res)))
+                        if dictFormat:
+                            os.write(stdout_fd, msgpack.packb({b'i': msgid, b'r': res}))
+                        else:
+                            os.write(stdout_fd, msgpack.packb((1, msgid, None, res)))
```

Comment:
maybe only if/else the arg for packb and only have 1 os.write(stdout_fd, msgpack.packb(arg))?